### ISSUE TITLE
check for invalid licenses?

### ISSUE THREAD

**Issue by dtenenba** [Bioconductor/BiocCheck#30]
@mrjc42 points out that people sometimes submit packages with (for example):

    License: GPL (>= 2)

Which is not valid because GPL is not versioned. GPL-2 is something totally different than GPL, IIUC.

However, version specifications in the License field are not always invalid, see http://cran.r-project.org/doc/manuals/r-release/R-exts.html#Licensing .

So the logic to determine whether to mark a license as invalid could get quite complicated, and would be subject to change over time. Not sure if I want to 'fix' this. But thought it was worth filing an issue about at least.

**Comment by nhayden** [comment]
Maybe I misunderstand, but I think `License: GPL (>= 2)` is given as an example of a standardized license specification in R-exts 1.1.2

**Comment by dtenenba** [comment]
You're right. If R-exts says it's valid, I guess it is. According to [the wikipedia page for GPL](https://en.wikipedia.org/wiki/GNU_General_Public_License), GPL-1 -2 -3 are actually successive versions of the same license.

**Comment by mrjc42** [comment]
So GPL-2 is effectively the same as GPL-3?

**Comment by dtenenba** [comment]
No, they are different: https://en.wikipedia.org/wiki/GNU_General_Public_License#Version_2 https://en.wikipedia.org/wiki/GNU_General_Public_License#Version_3

**Comment by mrjc42** [comment]
Wait really?  So then if you specify GPL>=2, which license does it specifically refer to:  GPL-2 or GPL-3?

**Comment by dtenenba** [comment]
I guess you are saying either one. Which brings us back to the original problem. But R-exts allows this, so I hesitate to make it a bad thing. Maybe a NOTE. I'll reopen.

**Comment by mrjc42** [comment]
Well I really am just asking questions.  I am not a lawyer so I don't really know what will happen if it ever ends up in a courtroom.  I just worry that in order to be protected by a license you would presumably have to specify an actual specific license (as opposed to a range of potential licenses).  

So if saying 'GPL-2' was actually equivalent to saying 'GPL3' (IOW if it just refers to the 'latest' version and that version is kept somewhere else), then I think it would be OK since the meaning of GPL>2 would then clearly mean 'the latest version of GPL'.  

BUT: if GPL-2 really is a different meaning than GPL-3 then I worry there may be a problem (and that seems to be the case?)?

Incidently, this question goes all the way back to Robert who initially taught me to review packages (and told me to make people choose when he saw this kind of license).  But that doesn't make him correct as Robert is not a lawyer either, and so this might really not be necessary, or it might only apply to "some" licenses.  :P

But unless we can be 100% certain that it's just noise to talk about ">=" clauses in licenses, I do think people should probably be told something about it.

**Comment by nhayden** [comment]
[This matrix](https://fedoraproject.org/wiki/Licensing:Main?rd=Licensing#GPLCompatibilityMatrix) includes the "I want to release a project under GPL-X" coupled with "I want to (use code | link to a library) from a project licensed under GPL-X" and it has GPL-X 'or later' clauses

**Comment by dtenenba** [comment]
tl;dr

**Comment by mrjc42** [comment]
OMG that matrix is AWFUL.  Seeing that makes we almost wonder if we should be allowing people to use GPL at all.  That's probably not ever going to be a practical option, but what a cluster.
